packages/api/src/hooks/useRemoteConfig.ts

```diff
@@ -0,0 +1,23 @@
+import { useQuery } from '@tanstack/react-query';
+import initData from '../features.json';
+
+const remoteConfigQuery = async function () {
+    if (!process.env.REMOTE_CONFIG_URL) {
+        throw new Error('Remote Config URL is not set!');
+    }
+    const response = await fetch(process.env.REMOTE_CONFIG_URL);
+    if (!response.ok) {
+        throw new Error('Remote Config Server is out of reach!');
+    }
+    return response.json();
+};
+
+function useRemoteConfig() {
+    return useQuery({
+        queryKey: ['remoteConfig'],
+        queryFn: remoteConfigQuery,
+        initialData: initData,
+    });
+}
```

Comment:
useRemoteConfig -- does not seem recurring. So, you will able to get a new config only after the page is refreshed. Didn't we want it to be real-time like in growthbook? How often user reloads deriv-app?

Reply:
the cache will be renewed after 5 mins.

Reply:
anyways I can change it to be less even. its configurable.

Reply:
It will be downloaded after 5 mins if user doesn't refresh the page.
if user refresh the page it will download a fresh one immediately

Reply:
Can you please point me to the code about 5 minutes?

Reply:
@ali-hosseini-deriv  so refetch after 5 minutes is the desired behaviour? you can add staleTime to 5 minutes to refetch it every 5 minutes

```
staleTime: 3000
```

Reply:
TBH - I'm not a fan of refetching stuff every so often. 
As it might trigger re-rendering of page or its elements at random and increases race hazards out there.

At least until we gonna establish really good patterns for rendering / updates etc, I would be keen on just not reloading it until page is refreshed (easy, predictable, won't wreck havoc in application updating/rerendering at random)

Reply:
So unless there are business requirements to have hot-reloading, IMO would be better to disable automatic refetch/refresh (like it happend with balance background refresh causing whitescreen every couple updates)

Reply:
I agree @wojciech-deriv. If it's really needed, another option could be to add a banner "the site has been updated, refresh to get the latest changes".

Reply:
At deriv.com we handle thousands of hundreds of users at a time, and I do not want any of them to have any chance to get into the entry point that was already disabled. So, recurring background checks are very valuable from my standpoint, and I would like to have this at least as an option.

> Add a banner "the site has been updated, refresh to get the latest changes"
This is a good option, but it will be also based on recurring checks.

Reply:
> recurring checks.

I personally don't think the recurring checks is a problem, it's the flashing/updating of the UI randomly that @wojciech-deriv was bringing up.

Reply:
yeah, basically, at the moment, we have not established a good patterns on application rendering / updating flow. Its completely random.

Gotcha is, that update in pretty much any component, can trigger reauthorisation. And that can trigger pretty much any logic anywhere in app and blow up everything at random (like it used to happen with wallets when seemingly innocent balance update was causing application to go blank at random :D)

Ultimately - I strongly recommend against anything which does not happen in predictable manner. Been there already with deriv-app. Have fun catching elusive bugs which happens only in certain unlucky/rare sequence of events.

Reply:
I think the flow should be: we turn of specific UI (e.g. whatsapp) and then we do actually turn of endpoint couple hours later @dan-deriv 

I do not think our frontend tech is ready at the moment to accomodate for such dynamic requirements like chaning UI-logic in runtime without page reload.

Reply:
I acknowledge the flashing/rendering problem. But it's the responsibility of the client who uses this hook. In deriv-com we can do this, and already tested it with Growthbook flags.

Reply:
as long as by default its turned off, to make sure its not gonna be overused.
I spend quite a lot of time tracking and fixing really random crashes and race hazards thats why I'm somewhat sensitive to anything which is not executed in predictable manner.

Reply:
@yashim-deriv the default behaviour of refetch in react-query is 5 mins by default.
I kept it untouched for now until we get the requirement from business perspective.

Reply:
really? all the hooks do refresh the data every 5 mins and cause random re-renders every 5 mins?

Reply:
Based on [react-query docs](https://tanstack.com/query/v4/docs/framework/react/guides/caching) the cache renews every 5 minutes or after switch back to the tab.
<img width="980" alt="image" src="https://github.com/binary-com/deriv-app/assets/81898967/30172957-bdd1-4d57-b1ab-812bac09df15">